### PR TITLE
Drop unaggregatable results from YCSB samples.

### PR DESCRIPTION
**Pull request by cmccoy**

### CODE REVIEW

File: perfkitbenchmarker/packages/ycsb.py [hunk 1]
```diff
@@ -345,19 +350,26 @@ def CombineHistograms(hist1, hist2):
...
+        logging.warn('"%s" in new dict, but not in old.', group_name)
+        result['groups'][group_name] = copy.deepcopy(group)
+        continue
       for k, v in group['statistics'].iteritems():
-        if k not in result['groups'][group_name]['statistics']:
-          logging.warn('"%s" in new dict, but not in old.', k)
-          result['groups'][group_name]['statistics'][k] = copy.deepcopy(v)
-          continue
-        elif k not in operators:
+        if k not in operators:
```
Comment (yuyantingzero) [inline perfkitbenchmarker/packages/ycsb.py]: Consider adding some inline comments on the logic here on the flow. It looks a bit confusing why you move the code around.

File: perfkitbenchmarker/packages/ycsb.py [hunk 2]
```diff
@@ -336,6 +334,13 @@ def _CombineResults(result_list, combine_histograms=True):
       'MinLatency(ms)': min,
```
Comment (yuyantingzero) [inline perfkitbenchmarker/packages/ycsb.py]: Add some inline comments on why this result will be dropped.

File: perfkitbenchmarker/packages/ycsb.py (hunk 1 above)
Comment (yuyantingzero) [inline perfkitbenchmarker/packages/ycsb.py]: Never mind, understand the flow now. But it is still good to add some inline comments or break the method into smaller ones. Feel free to ignore if only I feel confused about it.

File: perfkitbenchmarker/packages/ycsb.py [hunk 3]
```diff
@@ -318,8 +318,6 @@ def _CombineResults(result_list, combine_histograms=True):
   Returns:
     A dictionary, as returned by ParseResults.
   """
-  result = copy.deepcopy(result_list[0])
-
   # Binary operators to aggregate reported statistics.
   # 'None' will be dropped.
   operators = {
```
Comment (yuyantingzero) [inline perfkitbenchmarker/packages/ycsb.py]: I think this should be module level constant.

Comment (cmccoy) [inline perfkitbenchmarker/packages/ycsb.py]: done.

File: perfkitbenchmarker/packages/ycsb.py (hunk 2 above)
Comment (cmccoy) [inline perfkitbenchmarker/packages/ycsb.py]: done.

File: perfkitbenchmarker/packages/ycsb.py (hunk 1 above)
Comment (cmccoy) [inline perfkitbenchmarker/packages/ycsb.py]: Added a comment describing the flow.

**Comment by cmccoy** [comment]
PTAL.

**Comment by yuyantingzero** [comment]
LGTM

**Pull request merged**